models/Collection.cfc

```diff
@@ -409,6 +409,21 @@ component accessors="true" {
             return acc;
         }, {} );
     }
+    public struct function groupByUnique( required string key, boolean forceLookup = false ) {
+        return this.reduce( function( acc, item ) {
+            if ( ( isObject( item ) && structKeyExists( item, "get#key#" ) ) || forceLookup ) {
+                var value = invoke( item, "get#key#" );
+            }
+            else {
+                var value = item[ key ];
+            }
+            if ( ! structKeyExists( acc, value ) ) {
```

Comment:
I don't think you need these three lines if this is returning a struct.

Reply:
By aliasing...n/a anymore